### PR TITLE
feat(shipper): add Deutsche Post and DPD, updated shippers for german language and general fixes for shippers

### PR DESCRIPTION
**Pull request by Pr0mises**
## Proposed change
- Adding Deutsche Post Briefankündigung
- Adding DPD (kept dpd_com_pl, so no breaking change)
- Added/fixed german language to multiple providers
- Fixed german language

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests
- [ ] Documentation update
- [x] Adds a new shipper
- [x] Update existing shipper

## Additional information

I'm not sure if the hermes body excludes non german subjects or not. If yes feel free to change it.

### CODE REVIEW

File: custom_components/mail_and_packages/const.py [hunk 1]
```diff
@@ -982,10 +1082,12 @@
...
     "inpost_pl",
     "dpd_com_pl",
+    "dpd",
     "gls",
     "dhl_parcel_nl",
     "bonshaw_distribution_network",
     "purolator",
     "intelcom",
     "post_nl",
+    "post_de",
```
Comment (firstof9) [inline custom_components/mail_and_packages/const.py:1092]:
```suggestion
```

We only put shippers here that have both `delivering` and `delivered` sensors.

**Review by firstof9 — CHANGES_REQUESTED** [review]

**Comment by firstof9** [comment]
Looks like some tests will need updating on this.

File: custom_components/mail_and_packages/const.py (hunk 1 above)
Comment (Pr0mises) [reply custom_components/mail_and_packages/const.py:1092]: I thought so, got a bit confused because of Poczta post, as it is included without delivered

**Comment by Pr0mises** [comment]
Probably, I've no hass instance for checking against the tests in vscode rn (vm crashed). I only tested it on my production server

File: custom_components/mail_and_packages/const.py [hunk 2]
```diff
@@ -275,11 +277,20 @@
...
+            "parcel is now with your local Hermes courier",
+            "Ihre Hermes Sendung",
+        ],
+        "body": [
+            "Voraussichtliche Zustellung",
+        ],
     },
     "hermes_packages": {},
-    "hermes_tracking": {"pattern": ["\\d{16}"]},
+    "hermes_tracking": {"pattern": ["\\d{11,16}"]},
```
Comment (firstof9) [inline custom_components/mail_and_packages/const.py:293]: Are you sure the tracking number can be 11 or 16 digits?

Comment (Pr0mises) [reply custom_components/mail_and_packages/const.py:293]: According to https://www.myhermes.de/empfangen/sendungsverfolgung/

Between 11 and 20 digits

**Comment by firstof9** [comment]
Thanks!

**Pull request merged**